sound/soc/codecs/rt1308-sdw.c

```diff
@@ -561,13 +561,19 @@ static int rt1308_sdw_hw_params(struct snd_pcm_substream *substream,
 		return -EINVAL;
 	}
 
+	num_channels = params_channels(params);
+	ch_mask = (1 << (num_channels)) - 1;
+	if (rt1308->slots) {
+		num_channels = rt1308->slots;
+		ch_mask = rt1308->rx_mask;
+	}
```

Comment:
I'd use an "else" here - why would you assign two variables to immediately overwrite them?

Reply:
fixed